### PR TITLE
Pending requests api

### LINKED ISSUE #14: [4pts] Create Pending Requests Api (admin)
- a GET request to get all Requests only for admins
- a POST request to ACCEPT/DENY the Requests  that are not yet approved only for admin
- Save Notification

### PR DESCRIPTION
**Pull request by ichala**
### PENDING REQUESTS API : 
**************************

1. ### `GET /admin/requests/pending` : 
  Return All Pending Reservations
 **_example of response :_** 

``` JSON
    [
  {
        "id": 3,
        "hall_id": 1,
        "reserve_date": "2022-12-21",
        "user_id": 4,
        "status": "Pending",
        "user": {
           ...
        },
        "hall": {
           ...
        }
    },
    {
        "id": 4,
        "hall_id": 1,
        "reserve_date": "2022-10-30",
        "user_id": 5,
        "status": "Pending",
        "user": {
           ...
        },
        "hall": {
           ...
        }
    },
]
```

2. ### `POST /admin/requests/pending`
  **Required Params** : 

```JS
{
decision: "Confirm/Refuse",
reservation_id: "RESERVATION ID HERE",
text : " TEXT_HERE"
}
```

> if the decision is Confirmed the reservation status will be updated to 'Confirmed'
> if the decision is Refuse the reservation status will be updated to 'Refused'
> after any update, we add a notification to the notifications table

Closes https://github.com/ichala/short-let-api/issues/14

### CODE REVIEW

**Review by d2ndjim — APPROVED** [review]
Hi, @ichala , Great Job you've done on the project.
You can go ahead and merge the project now

68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6433317732347073475965656b435a792f67697068792e676966

## Highlights
✅ Good pull request description 👏 
✅ App has the correct functionality 🔥 
✅ Good commit messages 👍 

Cheers and Happy coding!👏👏👏

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -1,2 +1,63 @@
...
+  # If Admin Refuse Update reservation status
+  def refuse_reservation(reservation)
+    reservation.status = 'Refused'
+    if reservation.save
+      create_notification(reservation)
+      render json: reservation, each_serializer: ReservationSerializer
+    else
+      render json: { error: 'Bad Request' }, status: :not_acceptable
+    end
+  end
```
Comment (aimalamiri) [inline app/controllers/reservations_controller.rb]: These two methods are almost the same except the `status`. What do you think about something like this?
```ruby
def update_reservation(reservation, status)
    reservation.status = status
    if reservation.save
      create_notification(reservation)
      render json: reservation, each_serializer: ReservationSerializer
    else
      render json: { error: 'Bad Request' }, status: :not_acceptable
    end
end
```

And then we can call it like this

```ruby
update_reservation(reservation, 'Refused')
```
and
```ruby
update_reservation(reservation, 'Confirmed')
```

**Review by aimalamiri — CHANGES_REQUESTED** [review]
Hi  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the this thread if something is not 100% clear.

**Review by LYANGEND — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hello @ichala , 

### For now your task is complete!  There is nothing else to say other than....good work 👍  
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**Review by EidHachem — APPROVED** [review]
Well done @ichala

68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f766d74786e78766556556f64472f67697068792e676966

**Review by aimalamiri — APPROVED** [review]
### **Hi 👋** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**